ttnn/cpp/ttnn/operations/eltwise/binary/device/binary_composite_op.cpp

```diff
@@ -380,4 +379,29 @@ Tensor _polyval(const Tensor& input_a, const std::vector<float>& coeffs, const s
     return final_tensor;
 }
 
+Tensor ExecuteGCD::invoke(const Tensor& input_a, const Tensor& input_b, const std::optional<MemoryConfig>& output_mem_config) {
+    Tensor input_a_abs = ttnn::abs(input_a);
+    Tensor input_b_abs = ttnn::abs(input_b);
+    Tensor a_gt_b = ttnn::gt(input_a_abs, input_b_abs);
+    Tensor min = ttnn::where(a_gt_b, input_b_abs, input_a_abs);
+    Tensor max = ttnn::where(a_gt_b, input_a_abs, input_b_abs);
+    a_gt_b.deallocate();
+    constexpr std::size_t max_iterations = 10;
+    for (std::size_t iteration = 0; iteration < max_iterations; ++iteration) {
+        Tensor isz = ttnn::eqz(min);
+        Tensor rem = ttnn::remainder(max, ttnn::where(isz, isz, min));
+        max = ttnn::where(isz, max, min);
+        min = rem;
+    }
+    return ttnn::abs(max);
```

Comment:
`return max;` should be sufficient since `input_a` and `input_b` already have their absolute value taken. Also, can you add the following comment to the code above when you get the chance?
```
// https://en.wikipedia.org/wiki/Lam%C3%A9%27s_theorem
// While 186 is the theoretical maximum iterations for numbers within the floating point range according to Lame's
// theorem, in practice when evaluating gcd of consecutive Fibonacci numbers coerced to floating point, the
// maximum number of iterations reached is only [X] because the remainder converges to 0 much more quickly. In
// addition, limited precision in bfloat16 format decreases support for input to the range [-1024, 1024]
```
Make sure to replace `[X]` with the number of iterations you determine after adding the test with gcd(987, 610) I suggested on Slack.

Reply:
sure @patrickroberts . Thank you